src/Formatting/Combinators.hs

```diff
@@ -0,0 +1,513 @@
+-- "[11, 7, 1, 3]"
+--
+-- Note: the type variable 'f' will almost always be 'Format r', so the type of this function can be thought of as:
+--
+-- @
+-- roundedTo :: (Integral i, RealFrac d) => Format r (i -> r) -> Format r (d -> r)
+-- @
+roundedTo :: (Integral i, RealFrac d, Functor f) => f (i -> r) -> f (d -> r)
+roundedTo = fmap (. round)
+
+-- | Take a fractional number and truncate it before formatting it as the given Format:
+--
+-- >>> format (truncatedTo int) 6.66
+-- "6"
+-- >>> format (list (truncatedTo int)) [10.66, 6.66, 1.0, 3.4]
+-- "[10, 6, 1, 3]"
+--
+-- Note: the type variable 'f' will almost always be 'Format r', so the type of this function can be thought of as:
+--
+-- @
+-- truncatedTo :: (Integral i, RealFrac d) => Format r (i -> r) -> Format r (d -> r)
+-- @
+truncatedTo :: (Integral i, RealFrac d, Functor f) => f (i -> r) -> f (d -> r)
+truncatedTo = fmap (. truncate)
+
+-- | Take a fractional number and ceiling it before formatting it as the given Format:
+--
+-- >>> format (ceilingedTo int) 6.66
+-- "7"
+-- >>> format (list (ceilingedTo int)) [10.66, 6.66, 1.0, 3.4]
+-- "[11, 7, 1, 4]"
+--
+-- Note: the type variable 'f' will almost always be 'Format r', so the type of this function can be thought of as:
+--
+-- @
+-- ceilingedTo :: (Integral i, RealFrac d) => Format r (i -> r) -> Format r (d -> r)
+-- @
+ceilingedTo :: (Integral i, RealFrac d, Functor f) => f (i -> r) -> f (d -> r)
+ceilingedTo = fmap (. ceiling)
+
+-- | Take a fractional number and floor it before formatting it as the given Format:
+--
+-- >>> format (flooredTo int) 6.66
+-- "6"
+-- >>> format (list (flooredTo int)) [10.66, 6.66, 1.0, 3.4]
+-- "[10, 6, 1, 3]"
+--
+-- Note: the type variable 'f' will almost always be 'Format r', so the type of this function can be thought of as:
+--
+-- @
+-- flooredTo :: (Integral i, RealFrac d) => Format r (i -> r) -> Format r (d -> r)
+-- @
+flooredTo :: (Integral i, RealFrac d, Functor f) => f (i -> r) -> f (d -> r)
+flooredTo = fmap (. floor)
+
+-- | Use the given lens to view an item, formatting it with the given formatter.
+--
+-- You can think of this as having the type:
+--
+-- @
+-- 'viewed' :: 'Lens'' s a -> Format r (a -> r) -> Format r (s -> r)
+-- @
+--
+-- >>> format (viewed _1 int) (1, "hello")
+-- "1"
+--
+-- This is useful when combined with the Monoid instance for Format, because it allows us to give a data structure as an argument only once, and deconstruct it with the formatters:
+--
+-- @
+-- data Person = Person
+--   { _personName :: Text
+--   , _personAge :: Int
+--   }
+-- makeLenses ''Person
+--
+-- me :: Person
+-- me = Person "Alex" 38
+--
+-- format ("The person's name is " % quoted (viewed personName text) % ", and their age is " <> viewed personAge int) me
+-- "The person's name is 'Alex', and their age is 38"
+-- @
+viewed :: ((a -> Const a b) -> s -> Const a t) -> Format r (a -> r) -> Format r (s -> r)
+viewed l = fmap (. (getConst . l Const))
```

Comment:
maybe also add a non-lens `view` or `viewing` function here with an accessor function argument? (I'm sure this exists, but it seems like a consistent thing to have)